docs/validate_doc_examples.py

```diff
@@ -77,15 +80,19 @@ def create_fe_file(snippet: CodeSnippet) -> pathlib.Path:
     return fe_snippet_path
 
 def run_snippet(path: pathlib.Path) -> 'subprocess.CompletedProcess[bytes]':
-    return subprocess.run(["cargo", "run", str(path), '--overwrite'])
+    return
 
 def validate_code_examples() -> None:
     TMP_SNIPPET_DIR.mkdir(exist_ok=True)
+    build = subprocess.run(["cargo", "build"])
+    if build.returncode == 1:
+        exit(1)
+
     for path in get_all_doc_files():
         for snippet in get_all_snippets_in_file(path):
             if not snippet.unique_name() in SKIP_LIST:
                 fe_snippet = create_fe_file(snippet)
-                compile_result = run_snippet(fe_snippet)
+                compile_result = subprocess.run(["./target/debug/fe", str(fe_snippet), '--overwrite', '-e', 'abi'])
```

Comment:
I guess we should either move this back in `run_snippet` or remove `run_snippet`